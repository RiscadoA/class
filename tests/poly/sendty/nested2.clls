proc main() {
  cut {
    sendty x(wait);
    send x(y. println("wait 1"); wait y; ());
    sendty x(close);
    recv x(y);
    par {
      println("close 1"); close y
      ||
      send x(y. println("close 2"); close y);
      recv x(y); println("wait 2"); wait y;
      wait x;
      ()
    }
    |x: recvty A; recv ~A; recvty B; send A; recv ~B; send B; close|
    recvty x(A);
    recv x(a);
    recvty x(B);
    send x(a);
    recv x(b);
    send x(b);
    close x
  }
};;
