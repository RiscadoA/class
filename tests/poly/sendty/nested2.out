close 1
wait 1
close 2
wait 2
